modules/db.ts

```diff
@@ -123,7 +123,42 @@ const setInitPurchasedMiddleware = {
   },
 };
 
+/** Use Web Crypto API to set ID */
+const uuidMiddleware = {
```

Comment:
I threw this middleware together after quickly browsing through the Web Crypto API docs  - which I got ahead of myself by using. 

It's a cool API, and I'm glad I played with it a little, but [Dexie and IndexDB offer different methods of setting ids right out-of-the-box](https://dexie.org/docs/Syncable/Dexie.Syncable.js#2-use-uuid-based-primary-keys-)

We need to switch to this.

Reply:
Did switch to this, but as noted in other comments, we still needed this API to generate the IDs on first go. I added this to the methods themselves (addItem and addList) rather than having another piece of middleware. 

I think IRL this approach with be incredibly prone to bugginess - and creating truly unique identifiers is much more involved. But we've accomplished what this issue set out to fix - IDs are not numbers any longer, but UUID strings.